src/cookies.rs

```diff
@@ -1,25 +1,109 @@
 //! Cookie-related functionality for WebDriver.
 use serde::{Deserialize, Serialize};
 use time::OffsetDateTime;
-use webdriver::command::WebDriverCommand;
+use webdriver::command::{AddCookieParameters, WebDriverCommand};
+use webdriver::common::Date;
 
 use crate::client::Client;
 use crate::error;
 
 /// Type alias for a [cookie::Cookie]
 pub type Cookie<'a> = cookie::Cookie<'a>;
 
+/// Wrapper for serializing AddCookieParameters.
+#[derive(Debug, Serialize)]
+pub struct AddCookieParametersWrapper<'a> {
+    /// The cookie to serialize.
+    #[serde(with = "AddCookieParameters")]
+    pub cookie: &'a AddCookieParameters,
+}
+
+/// The SameSite parameter for cookies.
+///
+/// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
+/// for more details.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub(crate) enum SameSite {
+    /// Cookies will only be sent in a first-party context and not be sent along with
+    /// requests initiated by third party websites.
+    Strict,
+    /// Cookies are not sent on normal cross-site subrequests (for example to load images
+    /// or frames into a third party site), but are sent when a user is navigating to the
+    /// origin site (i.e., when following a link).
+    Lax,
+    /// Cookies will be sent in all contexts, i.e. in responses to both first-party and
+    /// cross-origin requests. If SameSite=None is set, the cookie Secure attribute must
+    /// also be set (or the cookie will be blocked).
+    None,
+}
+
+impl From<String> for SameSite {
+    fn from(s: String) -> Self {
+        match s.to_lowercase().as_ref() {
+            "strict" => SameSite::Strict,
+            "lax" => SameSite::Lax,
+            _ => SameSite::None,
```

Comment:
This clause makes me think this should be `TryFrom` (or `FromStr`), and not just `From`.

Reply:
Removed `SameSite`